### PR TITLE
Expose runtime docs on the released docs

### PR DESCRIPTION
**Pull request by wilzbach**
The DRuntime docs have been exposed for quite a while after https://github.com/dlang/dlang.org/pull/2084.

However, as there has been almost no activity in improving them, I think the only way to move forward is to expose them on the main pages too.
At the moment, `rt` is only visible on the `-prerelease` menu.
Example: https://dlang.org/phobos-prerelease/rt_config.html

We did the same with the DMD docs as they also received no attention while they were only available in the `-prerelease` for a year.

### CODE REVIEW

**Comment by CyberShadow** [comment]
It would be nice if the package could somehow have a description, or we could somehow visually separate public and internal APIs...

**Comment by wilzbach** [comment]
> It would be nice if the package could somehow have a description

We could add a `package.d` to `druntime/src/rt`?
(though I think currently the navigation doesn't support package.d files in the first-level, only second and lower levels)

> or we could somehow visually separate public and internal APIs...

Maybe a simple <hr> before <dmd> would do?
A bit fancier (and more work) would be to
- to use different folders for the DMD and runtime documentation (it never really made sense to put it under /phobos)
- have toggle buttons instead of "Library Reference" e.g. "Library" and "Internal". With internal being gray/hidden by default, but with its state saved in HTML5 WebStorage...

**Comment by CyberShadow** [comment]
> Maybe a simple `<hr>` before `<dmd>` would do?

Can we do that easily? Because if we can, then we can also add "Internal API:" below it, which would be perfect.

> to use different folders for the DMD and runtime documentation (it never really made sense to put it under /phobos)

That would work.

> have toggle buttons instead of "Library Reference" e.g. "Library" and "Internal". With internal being gray/hidden by default, but with its state saved in HTML5 WebStorage...

I don't like this. If we want to pull in JS, it's better to just separate the two categories visually. There's certainly enough space in the sidebar there.

**Comment by wilzbach** [comment]
> Can we do that easily? Because if we can, then we can also add "Internal API:" below it, which would be perfect.

Yeah, it would as easy as patching `modlist.d`:

```diff
diff --git a/tools/modlist.d b/tools/modlist.d
index 033d6601..0feda7c0 100644
--- a/tools/modlist.d
+++ b/tools/modlist.d
@@ -138,6 +138,9 @@ int main(string[] args)
     writeln("MODULE_MENU=");
     foreach (part; packages)
     {
+        if (part == "dmd")
+            writeln("$(HR) Internal API: $(BR)$(BR)");
+
         // check whether it's a package or file
         auto subTree = tree[part];
         if (subTree.leaves.length)
```

Though we should probably not go with this quick & dirty solution as `modlist.d` is used by a few other projects, e.g. mir
Hence I added dedicated CSS classes (not too much more work and a lot cleaner).
Preview:

image

File: css/style.css [hunk 1]
```diff
@@ -2120,3 +2120,17 @@ dt.d_decl:hover .decl_anchor {
 .message-box-green {
     background-color: #4EBA0F;
 }
+
+/**
+Custom modifications to the navigation menu
+- Insert an Internal API divider before the dmd menu item
+*/
+.modlist-submenu-dmd::before{
```
[the remark this answers is not in the document]
Comment (wilzbach) [reply css/style.css]: Yep, only CSS classes for the top-level items are generated.

**Pull request merged**